server/main_server/src/agent_main.cpp

```diff
@@ -0,0 +1,1326 @@
+
+            disconnectAllSvrToSvrConn();
+        }
+
+        if (irods::KW_CFG_SERVICE_ROLE_PROVIDER == svc_role) {
+            disconnectRcat();
+        }
+
+        irods::re_plugin_globals->global_re_mgr.call_stop_operations();
+    } // cleanup
+
+    auto cleanupAndExit(int status) -> void
+    {
+        cleanup();
+
+        if (status >= 0) {
+            std::exit(0); // NOLINT(concurrency-mt-unsafe)
+        }
+
+        std::exit(1); // NOLINT(concurrency-mt-unsafe)
+    } // cleanupAndExit
+
+    auto set_rule_engine_globals(RsComm& _comm) -> void
+    {
+        irods::set_server_property<std::string>(irods::CLIENT_USER_NAME_KW, _comm.clientUser.userName);
+        irods::set_server_property<std::string>(irods::CLIENT_USER_ZONE_KW, _comm.clientUser.rodsZone);
+        irods::set_server_property<int>(irods::CLIENT_USER_PRIV_KW, _comm.clientUser.authInfo.authFlag);
+        irods::set_server_property<std::string>(irods::PROXY_USER_NAME_KW, _comm.proxyUser.userName);
+        irods::set_server_property<std::string>(irods::PROXY_USER_ZONE_KW, _comm.proxyUser.rodsZone);
+        irods::set_server_property<int>(irods::PROXY_USER_PRIV_KW, _comm.clientUser.authInfo.authFlag);
+    } // set_rule_engine_globals
+
+    auto agentMain(RsComm& _comm) -> int
+    {
+        int status = 0;
+
+        // compiler backwards compatibility hack
+        // see header file for more details
+        irods::dynamic_cast_hack();
+
+        while (status >= 0) {
+            if (g_terminate) {
+                log_agent::trace("{}: Received shutdown instruction. Agent is shutting down.", __func__);
+                break;
+            }
+
+            // set default to the native auth scheme here.
+            if (!_comm.auth_scheme) {
+                _comm.auth_scheme = strdup("native");
+            }
+
+            // The following is an artifact of the legacy authentication plugins. This operation is
+            // only useful for certain plugins which are not supported in 4.3.0, so it is being
+            // left out of compilation for now. Once we have determined that this is safe to do in
+            // general, this section can be removed.
+#if 0
+            // construct an auth object based on the scheme specified in the comm
+            irods::auth_object_ptr auth_obj;
+            if (const auto err = irods::auth_factory(_comm.auth_scheme, &_comm.rError, auth_obj); !err.ok()) {
+                irods::experimental::api::plugin_lifetime_manager::destroy();
+
+                irods::log(PASSMSG(fmt::format(
+                    "Failed to factory an auth object for scheme: \"{}\".",
+                    _comm.auth_scheme), err));
+
+                return err.code();
+            }
+
+            irods::plugin_ptr ptr;
+            if (const auto err = auth_obj->resolve(irods::AUTH_INTERFACE, ptr); !err.ok()) {
+                irods::experimental::api::plugin_lifetime_manager::destroy();
```

Comment:
@alanking This is what I was thinking of when I pinged you earlier.

I think we just open an issue to consider removing this in iRODS 5. I don't think this will ever be re-enabled.

Thoughts?

Reply:
This is safe to remove. There is a similar block in the database plugin which I've already removed in one of my branches for something that I'm working on. Personally, I creating an issue for it is optional since it has been inactive since at least 4.3.1. I believe the comment is suggesting that it is only used by plugins such as GSI and Kerberos which have never seen a 4.3 release, and I imagine will not see a 5.0 release either.

Reply:
Gotcha. Will remove this.

Reply:
Done.